vinhloc30796/creativecontact-app
# feat(profile): allow users to edit profile & edit portfolio

## Description

Implements a comprehensive portfolio management system with internationalization support, including project editing capabilities, artwork asset management, and enhanced form state handling.

## Key Changes

1. Add portfolio artwork database schema and management system
2. Implement PortfolioEditForm component with i18n support
3. Create new API endpoints for portfolio and artwork assets
4. Enhance form state management with proper React hooks implementation
5. Add multi-project support with tabs interface

## Breaking Changes

- Portfolio project structure now requires assets API integration
- Database schema requires new migration for portfolio_artworks
- FormStateNav component now requires title prop
- BackButton component relocated from profile/edit to profile directory

## Related Issues

- Closes #WEB-211

## Testing Instructions

1. Navigate to profile page and access portfolio management
2. Add a new project and verify form validation
3. Upload media assets and check preview functionality
4. Test project editing and updates
5. Verify proper i18n display in both English and Vietnamese

## Additional Notes

- This PR includes significant database changes requiring migration
- Implements proper React hooks patterns for form state management
- Enhances UI/UX with responsive design and proper navigation
